connector/spanmetricsconnector/internal/metrics/metrics.go

```diff
@@ -259,11 +277,20 @@ func (m *SumMetrics) GetOrCreate(key Key, attributes pcommon.Map) *Sum {
 	return s
 }
 
-func (s *Sum) AddExemplar(traceID pcommon.TraceID, spanID pcommon.SpanID, value float64) {
-	e := s.exemplars.AppendEmpty()
-	e.SetTraceID(traceID)
-	e.SetSpanID(spanID)
-	e.SetDoubleValue(value)
+func (s *Sum) AddExemplar(traceID pcommon.TraceID, spanID pcommon.SpanID, value float64, maxCount int) {
```

Comment:
Please add test cases to cover cases where len of exemplars is less than the configured max count, and also is > configured max count.